python/cuda_cccl/tests/parallel/test_scan.py

```diff
@@ -58,6 +58,12 @@ def op(a, b):
         return a + b
 
     dtype = input_array.dtype
+
+    if dtype == np.float16:
+        reduce_op = parallel.OpKind.PLUS
+    else:
+        reduce_op = op
+
```

Comment:
In a separate PR, I think we need to expand the test so that all built-in types are tested with `reduce_op = parallel.OpKind.PLUS` and all but `float16` are also tested with a user-defined `op`.